Assets/_BForBoss/_Core/Scripts/WaveSystem/WaveModel.cs

```diff
@@ -1,71 +1,96 @@
+        private int DEBUG_ACTIVE_SPAWN_COUNT;
+
         private int _waveNumber;
         private int _initialMaxEnemyCount;
         private int _maxEnemyCount;
         private int _killCount;
+        private int _totalRoundTotalSpawnCount;
 
         public Action OnEnemySpawned;
-        public Action<int> OnEnemyKilled;
-        public Action<int, int> OnWaveCountUpdated;
-        
-        public int WaveNumber
+        public Action OnEnemyKilled;
+        public Action<int, int> OnDataUpdated;
+
+        private int WaveNumber
         {
             get => _waveNumber;
-            private set => _waveNumber = value;
+            set
+            {
+                _waveNumber = value;
+                OnDataUpdated?.Invoke(_waveNumber, _maxEnemyCount - _killCount);
+            }
         }
 
-        public int MaxEnemyCount
+        private int KillCount
         {
-            get => _maxEnemyCount;
-            private set => _maxEnemyCount = value;
+            get => _killCount; 
+            set
+            {
+                _killCount = value;
+                OnDataUpdated?.Invoke(_waveNumber, _maxEnemyCount - _killCount);
+            }
         }
 
-        public int KillCount
+        private int RoundTotalSpawnCount
         {
-            get => _killCount;
-            private set
+            get => _totalRoundTotalSpawnCount;
+            set
             {
-                _killCount = value;
-                OnEnemyKilled?.Invoke(_maxEnemyCount - _killCount);
+                _totalRoundTotalSpawnCount = value;
+                OnEnemySpawned?.Invoke();
             }
-            
         }
 
+        public bool IsMaxEnemySpawnedReached => _totalRoundTotalSpawnCount >= _maxEnemyCount;
+        public bool IsRoundConcluded => _maxEnemyCount == KillCount;
+
         public void SetupInitialWave(int maxEnemyCount)
         {
             _initialMaxEnemyCount = maxEnemyCount;
-            IncrementWave(maxEnemyCount);
+            _maxEnemyCount = maxEnemyCount;
+            IncrementWave(1);
         }
 
         public void IncrementSpawnCount()
         {
-            OnEnemySpawned?.Invoke();
+            DEBUG_ACTIVE_SPAWN_COUNT++;
+            RoundTotalSpawnCount++;
+            Debug.Log($"Enemy Spawned: <color=red>Active Spawn Count: {DEBUG_ACTIVE_SPAWN_COUNT}</color>");
+            Debug.Log($"Enemy Spawned: <color=red>Round Spawn Count {_totalRoundTotalSpawnCount}</color>");
+            Debug.Log($"===========================================");
         }
 
         public void IncrementKillCount()
         {
+            DEBUG_ACTIVE_SPAWN_COUNT--;
             KillCount++;
+            OnEnemyKilled?.Invoke();
+
+            Debug.Log($"Enemy Killed:  <color=red>Active Spawn Count: {DEBUG_ACTIVE_SPAWN_COUNT}</color>");
+            Debug.Log($"Enemy Killed:  <color=red>Round Spawn Count {_totalRoundTotalSpawnCount}</color>");
+            Debug.Log($"===========================================");
```

Comment:
I'm leaving these debug statements. It was very useful in debugging the spawn system, especially with all the Play Pause.